File: 12_MínimoComunMultiplo/Java/minComMul.java
import java.util.Scanner;
public class minComMul {
    public static void main(String[] args){
        Scanner leer=new Scanner(System.in);
        int i=2,c,mcm=1,k=1,nv;
        boolean v,v1;
        System.out.print("Ingresa numero de valores a evaluar: ");
        nv=leer.nextInt();
        int[] n=new int[nv];
        for(c=0;c<nv;c++){
            System.out.printf("Ingresa el valor %d: ",c+1);
            n[c]=leer.nextInt();
        }
        do{
            do{
                v=false; v1=false;
                for(c=0;c<nv;c++){
                    if(n[c]%i==0){
                        if(v==false){
                            System.out.printf("(%d)",i);
                            v=true;
                            k++;
                            mcm=mcm*i;
                        }
                        n[c]=n[c]/i;
                    }
                    else{
                        if(n[c]!=1)
                            v1=true;
                    }
                }
            }while(v==true);
            i=sigp(i);
        }while(v1==true);
        if(k==1)
            System.out.print("= 1\n");
        else
            System.out.printf("= %d\n",mcm);
    }
    public static boolean primo(int k){
        int i;
        for(i=2;i<k;i++){
            if(k%i==0)
                return false;
        }
        return true;
    }
    public static int sigp(int k){
        k++;
        while(!primo(k))
            k++;
        return k;
    }
}

File: 12_MínimoComunMultiplo/Python/minComMul.py
def primo(k):
    for i in range(2,k):
        if int(k%i)==0:
            return False
    return True
def sigp(h):
    h=int(h+1)
    while (not primo(h)):
        h=int(h+1)
    return h
i=int(2)
k=int(1)
mcm=int(1)
var=int(input("Ingresa numero de valores a evaluar: "))
n=[]
for c in range(var):
    n.append(int(input("Ingresa el valor %d: "%(c+1))))
while True:
    while True:
        v=False
        v1=False
        for c in range(var):
            if n[c]%i==0:
                if v==False:
                    print("(%d)"%(i))
                    k=k+1
                    mcm=mcm*i
                    v=True
                n[c]=n[c]/i
            else:
                if n[c]!=1:
                    v1=True
        if v==False:
            break
    i=sigp(i)
    if v1==False:
        break
if k==1:
    print("= 1")
else:
    print("= %d"%(mcm))